src/reim/reimZeroFilling.ts

```diff
@@ -8,7 +10,10 @@ import type { DataReIm } from '../types/index.ts';
  * truncated arrays if totalLength is smaller current length or
  * the same input if totalLength is equal that current length
  */
-export function reimZeroFilling(data: DataReIm, totalLength: number): DataReIm {
+export function reimZeroFilling<ArrayType extends DoubleArray>(
+  data: DataReIm<ArrayType>,
+  totalLength: number,
+): DataReIm<ArrayType | Float64Array<ArrayBuffer>> {
```

Comment:
There are still two possible array types in the return, so basically this only "works as expected" if the input has Float64Array in it, right?